TerrorConsole/Assets/_Main/Scripts/LevelManager/LevelManager.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using UnityEngine;
+
+namespace TerrorConsole
+{
+    public class LevelManager : Singleton<ILevelManagerSource>, ILevelManagerSource
+    {
+
+        public event Action<LevelState> OnLevelStateChange;
+
+        [SerializeField] private LevelState currentLevelState;
+        
+        public enum LevelState
+        {
+            Play,
+            Pause,
+            InDialogue,
+            Cinematic
+        }
+
+        public void ChangeLevelState(LevelState newState)
+        {
+            currentLevelState = newState;
+            OnLevelStateChange?.Invoke(newState);
+        }
+
+        public LevelState GetCurrentState()
+        {
+            return currentLevelState;
+        }
+
+        [ContextMenu("DEBUG_ChangeState_Play")]
+        public void DEBUG_ChangeState_Play()
+        {
+            ChangeLevelState(LevelState.Play);
+        }
+
+        [ContextMenu ("DEBUG_ChangeState_Pause")]
+        public void DEBUG_ChangeState_Pause()
+        {
+            ChangeLevelState(LevelState.Pause);
+        }
```

Comment:
Move this to a partial class on this same script and encapsulate using "#if UNITY_EDITOR", more information DM me or search on the internet, you are going to have 2 classes with same name on this script but one will be for Editor ONLY.